packages/angular_devkit/build_angular/src/builders/browser-esbuild/schema.json

```diff
@@ -216,6 +216,11 @@
       "type": "string",
       "description": "The full path for the new output directory, relative to the current workspace.\nBy default, writes output to a folder named dist/ in the current project."
     },
+    "outExtension": {
```

Comment:
Following todays design review this should not be necessary as the option can be provided directly when run the bundle function.

Reply:
Agree that we could probably always generate `*.mjs` files. I expect many users won't like the "aesthetics" of it, but that's a pretty minor complaint. Another option is to move this flag to `InfrastructureSettings` to make it a little more "private", as I don't think it actually needs to be in the public schema. WDYT @clydin?

Reply:
My 2 cents would be to use always `.mjs` for simplicity. I definitely don't think that we should expose to the users. I see that this option can become confusing to us pretty soon in (16.1) when we want esbuilder to generate both `.js` and `.mjs` for the same run. (`.mjs` for the server and `.js` for the browser).

Reply:
I'd prefer we stick with `.js` since its the "standard" file extension for JavaScript files.  `.mjs` is a Node.js invention and is otherwise unrelated to the ECMAScript specification and browsers in general.
Changing the file extension default could have unknown effects on downstream tools and behavior (secondary processing, deployment, server configurations, mime-type entry setup, etc.).

Reply:
For an additional option, the build system could potentially be setup to not need to output `.mjs` files on the server. A `package.json` file containing the following will cause Node.js to execute ESM `.js` files appropriately:
```
{
  "type": "module"
}
```
This could be placed in the output directory of server/test code as an auto-generated asset as needed.

Reply:
Good points @clydin

Reply:
Moved `outExtension` to `InternalOptions` so it is hidden from public API, but left `*.js` as the default. `package.json` is definitely another viable option, though since it's also a Node-ism I'm not sure that's strictly better.

Hopefully this should be good enough for now.